components/aidokuWrapper.tsx

```diff
@@ -46,27 +52,57 @@ export default function AidokuWrapper() {
       if (e.target == null) {
         setConsoleOutput((consoleOutput) => [
           ...consoleOutput,
-          'ERROR: Something went wrong when parsing your backup, try uploading again.'
+          '> ERROR: Something went wrong when parsing your backup, try uploading again.'
         ])
         return
       }
 
-      const backupResult = Paperback.toAidoku(e.target.result as string)
-      const convertedBackup = backupResult.backup
+      let backupResult: AidokuResult;
+      if (event.target.id === 'uploadPaperback') {
+        backupResult = Paperback.toAidoku(new TextDecoder().decode(e.target.result as ArrayBuffer))
 
-      setAidokuJson(JSON.stringify(convertedBackup))
+        setAidokuJson(JSON.stringify(backupResult.backup))
+      } else if (event.target.id === 'uploadTachiyomi') {
+        const data = pako.inflate(e.target.result as ArrayBuffer)
+        backupResult = Tachiyomi.toAidoku(data)
+        
+        setAidokuJson(JSON.stringify(backupResult.backup, (_, v) => {
+          const date = Date.parse(v);
+          // Matches only what Date.toJSON() would return
+          if (isNaN(date) || !(v.constructor === String && v.match(/\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}Z/))) {
+            return v;
+          }
+          return Math.floor(date / 1000);
+        }))
```

Comment:
This weird replacer function is because I set most date properties to use the `Date` object, which serializes to an ISO8601 string; however, Aidoku expects that it's the number of seconds since Unix epoch.